### PR TITLE
docs: Update cuda installation and support docs

### PR DESCRIPTION
**Pull request by wckzhang**
Added updated documentation for the dso type cuda
support and the updated ofi mtl support.

Signed-off-by: William Zhang <wilzhang@amazon.com>

### CODE REVIEW

**Review by Akshay-Venkatesh — APPROVED** [review]

File: docs/networking/cuda.rst [hunk 1]
```diff
@@ -38,15 +38,30 @@ Open MPI offers two flavors of CUDA support:
...
 Open MPI should be configured using the ``--with-cuda=<path-to-cuda>``
-configure option to build CUDA support into Open MPI.
+and ``--with-cuda-libdir=<path-to-libcuda.so>`` configure options to
+build CUDA support into Open MPI.
 
-This affects the smcuda shared memory btl, as well as the uct btl.
+Open MPI supports building with cuda libraries and running on systems
+without cuda libraries or hardware. In order to take advantage of
+this functionality, when compiling, you have to specify the cuda
+dependent components to be built as dso's using the
```
Comment (jsquyres) [inline docs/networking/cuda.rst:53]:
```suggestion
Open MPI supports building with CUDA libraries and running on systems
without CUDA libraries or hardware. In order to take advantage of
this functionality, when compiling, you have to specify the CUDA
dependent components to be built as DOSs using the
```

File: docs/networking/cuda.rst [hunk 2]
```diff
@@ -38,15 +38,30 @@ Open MPI offers two flavors of CUDA support:
...
+``--enable-mca-dso=<comma-delimited-list-of-cuda-components.``
+configure option.
+
+This affects the smcuda shared memory btl, rcache rgpusm, rcache
+gpusm, accelerator cuda, as well as the uct btl.
```
Comment (jsquyres) [inline docs/networking/cuda.rst:58]:
```suggestion
This affects the ``smcuda`` shared memory and ``uct`` BTLs, as well as the ``rgpusm`` and ``gpusm`` rcache components.
```

File: docs/networking/cuda.rst [hunk 3]
```diff
@@ -38,15 +38,30 @@ Open MPI offers two flavors of CUDA support:
...
+
+An example configure command would look like the following:
+
+   .. code-block:: sh
+      # Configure Open MPI this way
+      shell$ ./configure --with-cuda=/usr/local/cuda --with-cuda-libdir=/usr/local/cuda/lib64/stubs --enable-mca-dso=btl-smcuda,rcache-rgpusm,rcache-gpusm,accelerator-cuda <other configure params>
```
Comment (jsquyres) [inline docs/networking/cuda.rst:64]: This is probably too wide.  Can you break it up into 2 lines, with a `\` continuation character?

File: docs/networking/cuda.rst [hunk 4]
```diff
@@ -152,6 +168,21 @@ For more information refer to the `Intel Omni-Path documentation
...
+---------------------
+
+CUDA-aware support is present in OFI MTL.  When running CUDA-aware
+Open MPI over Libfabric, the OFI MTL will check if there are any
+providers capable of handling gpu memory through the hmem related
+flags. If a cuda capable provider is available, the OFI MTL will
+directly send GPU buffers through libfabric's API after registering
+the memory. If there are no cuda capable providers available, the
+buffers will automatically be copied to host buffers before being
+transferred through libfabric's API.
```
Comment (jsquyres) [inline docs/networking/cuda.rst:181]:
```suggestion
providers capable of handling GPU (or other accelerator) memory through the ``hmem``-related
flags. If a CUDA-capable provider is available, the OFI MTL will
directly send GPU buffers through Libfabric's API after registering
the memory. If there are no CUDA-capable providers available, the
buffers will automatically be copied to host buffers before being
transferred through Libfabric's API.
```

**Review by jsquyres — CHANGES_REQUESTED** [review]

File: docs/networking/cuda.rst (hunk 4 above)
Comment (wckzhang) [reply docs/networking/cuda.rst]: Done

File: docs/networking/cuda.rst (hunk 3 above)
Comment (wckzhang) [reply docs/networking/cuda.rst]: Done

File: docs/networking/cuda.rst (hunk 2 above)
Comment (wckzhang) [reply docs/networking/cuda.rst]: Done

File: docs/networking/cuda.rst (hunk 1 above)
Comment (wckzhang) [reply docs/networking/cuda.rst]: Done

File: docs/networking/cuda.rst [hunk 5]
```diff
@@ -38,15 +38,30 @@ Open MPI offers two flavors of CUDA support:
...
+``--enable-mca-dso=<comma-delimited-list-of-cuda-components.``
+configure option.
+
+This affects the smcuda shared memory btl, rcache rgpusm, rcache
+gpusm, accelerator cuda, as well as the uct btl.
+
+An example configure command would look like the following:
+
+   .. code-block:: sh
+      # Configure Open MPI this way
```
[the remark this answers is not in the document]
Comment (wckzhang) [reply docs/networking/cuda.rst:64]: Done

**Review by jsquyres — APPROVED** [review]

**Pull request merged**